luarules/gadgets/unit_preaim.lua

```diff
@@ -14,19 +14,25 @@ if not gadgetHandler:IsSyncedCode() then
 	return
 end
 
+--use customparams.exclude_preaim = true, to exclude units from being able to pre-aim at targets almost within firing range.
+--this is a good idea for pop-up turrets so they don't prematurely reveal themselves.
+--also when proximityPriority is heavily biased toward far targets
+
 local weaponRange = {}
 local isPreaimUnit = {}
 for unitDefID, unitDef in pairs(UnitDefs) do
-	if not unitDef.canFly then
+	if not unitDef.canFly and not unitDef.customParams.exclude_preaim then
 		local weapons = unitDef.weapons
 		if #weapons > 0 then
 			for i=1, #weapons do
-				if not isPreaimUnit[unitDefID] then
-					isPreaimUnit[unitDefID] = {}
+				if not WeaponDefs[weapons[i].weaponDef].customParams.exclude_preaim then
```

Comment:
Apply this to the weapons of units from the manual list below and remove the list. This lets the param be discovered "in the wild" by people looking at defs, and also makes sure the param is the actual source of truth and not getting overridden by the manual list.

Reply:
I would prefer to remove the gadget table entirely and just iterate for the customparams only. I will do that right away since it seems desired.